sonic_sku_create/sonic_sku_create.py

```diff
@@ -0,0 +1,329 @@
+			m = re.match(pattern,self.default_lanes_per_port[int(fp)-1])
+			if (splt == 1) :
+				self.portconfig_dict[idx_arr[0]][lanes_index] = m.group(1)+","+m.group(2)+","+m.group(3)+","+m.group(4) 
+				self.portconfig_dict[idx_arr[0]][index_index] = fp
+				self.portconfig_dict[idx_arr[0]][name_index] = "Ethernet"+str((int(fp)-1)*4)
+			elif (splt == 2) :
+				self.portconfig_dict[idx_arr[0]][lanes_index] = m.group(1)+","+m.group(2) 
+				self.portconfig_dict[idx_arr[1]][lanes_index] = m.group(3)+","+m.group(4)
+				self.portconfig_dict[idx_arr[0]][index_index] = fp 
+				self.portconfig_dict[idx_arr[1]][index_index] = fp
+				self.portconfig_dict[idx_arr[0]][name_index] = "Ethernet"+str((int(fp)-1)*4) 
+				self.portconfig_dict[idx_arr[1]][name_index] = "Ethernet"+str((int(fp)-1)*4+2)
+			elif (splt == 4) :
+				self.portconfig_dict[idx_arr[0]][lanes_index] = m.group(1) 
+				self.portconfig_dict[idx_arr[1]][lanes_index] = m.group(2) 
+				self.portconfig_dict[idx_arr[2]][lanes_index] = m.group(3) 
+				self.portconfig_dict[idx_arr[3]][lanes_index] = m.group(4)
+				self.portconfig_dict[idx_arr[0]][index_index] = fp 
+				self.portconfig_dict[idx_arr[1]][index_index] = fp 
+				self.portconfig_dict[idx_arr[2]][index_index] = fp 
+				self.portconfig_dict[idx_arr[3]][index_index] = fp
+				self.portconfig_dict[idx_arr[0]][name_index] = "Ethernet"+str((int(fp)-1)*4) 
+				self.portconfig_dict[idx_arr[1]][name_index] = "Ethernet"+str((int(fp)-1)*4+1) 
+				self.portconfig_dict[idx_arr[2]][name_index] = "Ethernet"+str((int(fp)-1)*4+2) 
+				self.portconfig_dict[idx_arr[3]][name_index] = "Ethernet"+str((int(fp)-1)*4+3)
+		self.platform_specific()	
+	
+	
+
+		
+	def create_port_config(self) :
+		#create a port_config.ini file based on the sku definition 
+		if not os.path.exists(self.new_sku_dir):
+			print ("Error - path:", self.new_sku_dir, " doesn't exist")
+			
+		try:
+			f = open(self.new_sku_dir+"port_config.ini","w+")
+		except IOError:
+			print ("Could not open file "+ self.new_sku_dir+"port_config.ini")
+		header = PORTCONFIG_HEADER # ["name", "lanes", "alias", "index"]
+		port_config = []
+		for line in self.portconfig_dict.values():
+			port_config.append(line)
+		 
+		port_config.sort(key=lambda x: (int(re.search(('\d+'),x[0]).group(0)))) # sort the list with interface name
+		f.write(tabulate(port_config, header,tablefmt="plain"))
+		f.close()
+	
+	def print_port_config(self) :
+		#print a port_config.ini file based on the sku definition 
+		
+		header = PORTCONFIG_HEADER # ["name", "lanes", "alias", "index"]
+		port_config = []
+		for line in self.portconfig_dict.values():
+			port_config.append(line)
+		 
+		port_config.sort(key=lambda x: (int(re.search(('\d+'),x[0]).group(0)))) # sort the list with interface name
+		print(tabulate(port_config, header,tablefmt="plain"))
+		
+		
+	def create_sku_dir(self) :
+		# create a new SKU directory based on the base SKU
+		try:
+			shutil.copytree(self.base_sku_dir, self.new_sku_dir)
```

Comment:
>copytree [](start = 10, length = 8)

If we are overwrite an existing sku dir in the image without `--remove`, we should report the error and exit.

Reply:
Added a check if the SKU exists, if yes USER should use the -r flag first